monkey/infection_monkey/system_info/windows_cred_collector/test_pypykatz_handler.py

```diff
@@ -0,0 +1,84 @@
+             'masterkey': '6e81d0cfd5e9ec083cfbdaf4d25b9cc9cc6b72947f5e80920034d1275d8613532025975ef051e891c30e6e9af6db54500fedfed1c968389bf6262c77fbaa68c9',
+             'sha1_masterkey': 'bbdabc3cd2f6bcbe3e2cee6ce4ce4cebcef4c6da', 'luid': 123086},
+            {'credtype': 'dpapi', 'key_guid': '9123-123ae123de4-121239-3123-421f',
+             'masterkey': '6e81d0cfd5e9ec083cfbdaf4d25b9cc9cc6b72947f5e80920034d1275d8613532025975ef051e891c30e6e9af6db54500fedfed1c968389bf6262c77fbaa68c9',
+             'sha1_masterkey': 'bbdabc3cd2f6bcbe3e2cee6ce4ce4cebcef4c6da', 'luid': 123086},
+            {'credtype': 'dpapi', 'key_guid': '9123-123ae123de4-121239-3123-421f'}],
+        'kerberos_creds': [
+            {'credtype': 'kerberos', 'username': 'monkey_kerb', 'password': None, 'domainname': 'ReAlDoMaIn',
+             'luid': 123086, 'tickets': []}],
+        'credman_creds': [
+            {'credtype': 'credman', 'username': 'monkey', 'domainname': 'monkey.ad.monkey.com',
+             'password': 'canyoufindme2', 'luid': 123086},
+            {'credtype': 'credman', 'username': 'monkey@monkey.com', 'domainname': 'moneky.monkey.com',
+             'password': 'canyoufindme1', 'luid': 123086},
+            {'credtype': 'credman', 'username': 'test', 'domainname': 'test.test.ts', 'password': 'canyoufindit',
+             'luid': 123086}],
+        'tspkg_creds': []}
+
+    def test__get_creds_from_pypykatz_session(self):
+        results = _get_creds_from_pypykatz_session(TestPypykatzHandler.PYPYKATZ_SESSION)
+
+        test_dicts = [{'username': 'monkey',
+                       'ntlm_hash': '31b73c59d7e0c089c031d6cfe0d16ae9',
+                       'password': '',
+                       'lm_hash': ''},
+                      {'username': 'monkey',
+                       'ntlm_hash': '',
+                       'password': 'canyoufindme',
+                       'lm_hash': ''},
+                      {'username': 'monkey123',
+                       'ntlm_hash': '',
+                       'password': 'canyoufindme123',
+                       'lm_hash': ''},
+                      {'username': 'monk3y',
+                       'ntlm_hash': '',
+                       'password': 'canyoufindm3',
+                       'lm_hash': ''},
+                      {'username': 'monkey',
+                       'ntlm_hash': '',
+                       'password': 'canyoufindme2',
+                       'lm_hash': ''},
+                      {'username': 'monkey@monkey.com',
+                       'ntlm_hash': '',
+                       'password': 'canyoufindme1',
+                       'lm_hash': ''},
+                      {'username': 'test',
+                       'ntlm_hash': '',
+                       'password': 'canyoufindit',
+                       'lm_hash': ''},
+                      ]
+        results = [result.to_dict() for result in results]
+        for test_dict in test_dicts:
+            self.assertTrue(test_dict in results)
```

Comment:
I might be mistaken but I think you can do this with `assert set(test_dicts) <= set(results)` which might be more efficient. Not critical at all, just a nice Pythonic way to do it 🐍

Reply:
Dicts are not hashable, so native dict can't be used in set. I converted it to even more pythonic way though :)